scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,980 @@
+------------------------------------
+-- Egg Hunt Egg-Stravaganza
+------------------------------------
+require("scripts/globals/zone")
+require("scripts/globals/utils")
+require("scripts/globals/npc_util")
+------------------------------------
+xi = xi or {}
+xi.events = xi.events or {}
+xi.events.egg_hunt = xi.events.egg_hunt or {}
+xi.events.egg_hunt.data = xi.events.egg_hunt.data or {}
+xi.events.egg_hunt.entities = xi.events.egg_hunt.entities or {}
+
+local event = SeasonalEvent:new("egg_hunt")
+
+-- Default settings
+local settings =
+{
+    START  = { DAY   =  6, MONTH = 4 },
+    FINISH = { DAY   = 17, MONTH = 4 },
+
+    VAR =
+    {
+        DAILY_EGG     = "[EGG_HUNT]DAILY_EGG",
+        DAILY_REWARD  = "[EGG_HUNT]DAILY_REWARD",
+        DAILY_BONUS   = "[EGG_HUNT]DAILY_BONUS",
+        DAILY_HELM    = "[EGG_HUNT]DAILY_HELM",
+        FIRST_THREE   = "[EGG_HUNT]FIRST_THREE",
+    },
+
+    -- Default era is 2005
+    ERA_2006 = false, -- Orphic Egg
+    ERA_2007 = false, -- Jeweled Egg and Egg Helm
+    ERA_2008 = false, -- Tier 2 nation eggs, allows trading Hard-Boiled Eggs
+    ERA_2009 = false, -- Egg Buffet set
+    -- 2009, 2010, 2011 and 2012 are identical
+    ERA_2013 = false, -- Prinseggstarta
+    ERA_2014 = false, -- Hatchling Shield, Copse Candy, Cracker
+    ERA_2015 = false, -- Rabbit Cap, show Rabbit Cap wearing NPCs
+    ERA_2018 = false, -- Allows trading Sairui-Ran x99 and Imperial Egg
+    ERA_2019 = false, -- Allows trading Apkallu Egg
+
+    -- Consolation prizes for repeating combinations where
+    -- the player has already received the relevant reward
+    MINOR_REWARDS = true,
+
+    -- Don't add words here
+    -- settings/main.lua will overwrite these
+    BONUS_WORDS = {},
+}
+
+local function loadSettings(currentTable, settingsName)
+    local settingTable = xi.settings.main[settingsName]
+
+    if not settingTable then
+        return
+    end
+
+    -- Load from main settings into current table
+    for settingName, _ in pairs(currentTable) do
+        if settingTable[settingName] then
+            currentTable[settingName] = settingTable[settingName]
+        end
+    end
+end
+
+loadSettings(settings, "EGG_HUNT")
```

Comment:
I'd probably add some temporary logging to debug this, and make sure this isn't firing multiple times on startup (it _shouldn't do_, but I get suspicious vibes from this)

Reply:
Added logging and confirmed it's loading once. Left it in as an option (Default: Off), so users can choose to see if their settings loaded from `settings/main.lua`.